mutablesecurity/modules/solutions_manager/solutions/teler.py

```diff
@@ -0,0 +1,453 @@
+            state=state,
+            host=host,
+            sudo=True,
+            name="Copy the default configuration file",
+            src=os.path.dirname(__file__) + "/../files/teler/teler.conf.j2",
+            dest="/opt/teler/teler.conf",
+            configuration=j2_configuration,
+        )
+
+        server.shell(
+            state=state,
+            host=host,
+            sudo=True,
+            name="Creates the log file",
+            commands=["touch /var/log/teler.log"],
+        )
+
+        teler_command = f"tail -f {Teler._configuration['log_location']} | /opt/teler/teler -c /opt/teler/teler.conf"
+        server.crontab(
+            state=state,
+            host=host,
+            sudo=True,
+            name="Adds a crontab to automatically run the binary",
+            command=teler_command,
+            present=True,
+            special_time="@reboot",
+        )
+        server.shell(
+            state=state,
+            host=host,
+            sudo=True,
+            name="Run the binary now, without a restart to trigger the crontab",
+            commands=[f"tmux new -d '{teler_command}'"],
+        )
+
+        Teler._put_configuration(state=state, host=host)
+
+        Teler.result = True
+
+    @deploy
+    def test(state, host):
+        Teler.get_configuration(state=state, host=host)
+
+        url = "http://localhost:" + str(Teler._configuration["port"])
+        wget_command = (
+            "wget --header 'User-Agent: curl x MutableSecurity' -O /tmp/index.html "
+            + url
+        )
+        server.shell(
+            state=state,
+            host=host,
+            sudo=True,
+            name="Requests the website with a bad user agent",
+            commands=[wget_command],
+        )
+
+        def stage(state, host):
+            # This verification will check if the last alert is one related to the
+            # request above.
+            alerts = host.get_fact(TestBadUserAgent)
+
+            Teler.result = alerts != 0
+
+        python.call(state=state, host=host, sudo=True, function=stage)
+
+    @deploy
+    def get_stats(state, host):
+        ManagedStats.get_stats(state=state, host=host, solution_class=Teler)
+
+    @deploy
+    def get_logs(state, host):
+        Teler.get_configuration(state=state, host=host)
+
+        Teler.result = host.get_fact(Logs)
+
+    @deploy
+    def update(state, host):
+        # Check if the latest release has a greater version than the local
+        # software
+        local_version = host.get_fact(Version)
+        github_latest_version = GitHub.get_latest_release_name("kitabisa", "teler")
+        if version.parse(local_version) == version.parse(github_latest_version):
+            Teler.result = True
+
+            return
+
+        # If the version on GitHub is greater, then update the software
+        Teler.uninstall(state=state, host=host)
+        Teler.install(state=state, host=host)
+
+    @deploy
+    def uninstall(state, host):
```

Comment:
What about the log files and the whole folder from `/opt/mutablesecurity/teler`? Shouldn't they also be deleted?

Reply:
Let's preserve the configuration files that we create for a specific solution (here `/opt/mutablesecurity/teler`). You may want to reinstall teler and keep the same configuration as in the previous installation. But yes, the log file that is set in the configuration (defaults to `/var/log/teler.log`) needs to be deleted.

Reply:
@AntociAlin, I adapted the uninstallation operation. Please recheck it.